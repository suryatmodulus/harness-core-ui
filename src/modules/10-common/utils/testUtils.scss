/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
