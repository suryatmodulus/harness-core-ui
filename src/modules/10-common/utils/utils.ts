/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
