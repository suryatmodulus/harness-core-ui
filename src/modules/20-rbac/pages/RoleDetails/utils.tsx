/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import produce from 'immer'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import type { Permission, PermissionResponse } from 'services/rbac'

export const getPermissionMap = (permissionList?: PermissionResponse[]): Map<ResourceType, Permission[]> => {
  const permissionsMap: Map<ResourceType, Permission[]> = new Map()
  permissionList?.map(response => {
    if (response.permission.resourceType) {
      if (permissionsMap.has(response.permission.resourceType as ResourceType)) {
        permissionsMap.get(response.permission.resourceType as ResourceType)?.push(response.permission)
      } else permissionsMap.set(response.permission.resourceType as ResourceType, [response.permission])
    }
  })
  return permissionsMap
}

export const onPermissionChange = (
  permission: string,
  isAdd: boolean,
  permissions: string[],
  setPermissions: (value: React.SetStateAction<string[]>) => void
): void => {
  if (isAdd) {
    if (
      permission === PermissionIdentifier.EDIT_DASHBOARD ||
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    ) {
      setPermissions(_permissions => [...permissions, permission, PermissionIdentifier.VIEW_DASHBOARD])
    } else {
      setPermissions(_permissions => [...permissions, permission])
    }
  } else if (
    !(
      permission === PermissionIdentifier.VIEW_DASHBOARD &&
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    )
  ) {
    setPermissions(_permissions =>
      produce(_permissions, draft => {
        draft?.splice(permissions.indexOf(permission), 1)
      })
    )
  }
}
