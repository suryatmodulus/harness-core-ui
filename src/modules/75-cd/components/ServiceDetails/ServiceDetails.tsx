/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout } from '@wings-software/uicore'
import { ServiceDetailsHeader } from '@cd/components/ServiceDetails/ServiceDetailsHeader/ServiceDetailsHeader'
import { ServiceDetailsContent } from '@cd/components/ServiceDetails/ServiceDetailsContent/ServiceDetailsContent'

const ServiceDetails: React.FC = () => {
  return (
    <Layout.Vertical>
      <ServiceDetailsHeader />
      <ServiceDetailsContent />
    </Layout.Vertical>
  )
}

export default ServiceDetails
