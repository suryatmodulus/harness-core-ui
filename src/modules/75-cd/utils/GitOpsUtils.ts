/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GitOpsInfoDTO, ManagedArgoGitOpsInfoDTO, ConnectedArgoGitOpsInfoDTO } from 'services/cd-ng'

export type GitOpsProviderType = Required<GitOpsInfoDTO>['type']

export const GitOpsProviderTypeEnum: Record<GitOpsProviderType, GitOpsProviderType> = {
  CONNECTED_ARGO_PROVIDER: 'CONNECTED_ARGO_PROVIDER',
  MANAGED_ARGO_PROVIDER: 'MANAGED_ARGO_PROVIDER'
}

export function isConnectedGitOpsProvider(obj?: GitOpsInfoDTO | null): obj is ConnectedArgoGitOpsInfoDTO {
  return obj?.type === GitOpsProviderTypeEnum.CONNECTED_ARGO_PROVIDER
}

export function isManagedGitOpsProvider(obj?: GitOpsInfoDTO | null): obj is ManagedArgoGitOpsInfoDTO {
  return obj?.type === GitOpsProviderTypeEnum.MANAGED_ARGO_PROVIDER
}
