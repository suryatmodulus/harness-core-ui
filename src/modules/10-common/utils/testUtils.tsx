/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { queryByAttribute } from '@testing-library/react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation, useHistory } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uicore'
import qs from 'qs'

import { enableMapSet } from 'immer'
import { AppStoreContext, AppStoreContextProps } from 'framework/AppStore/AppStoreContext'
import { LicenseStoreContext, LicenseStoreContextProps } from 'framework/LicenseStore/LicenseStoreContext'
import { LICENSE_STATE_VALUES } from 'framework/LicenseStore/licenseStoreUtil'
import { withAccountId, accountPathProps } from '@common/utils/routeUtils'
import type { Project } from 'services/cd-ng'
import { StringsContext } from 'framework/strings'

import { FeaturesContext, FeaturesContextProps } from 'framework/featureStore/FeaturesContext'
import type { FeatureDetail, FeatureMetaData } from 'framework/featureStore/featureStoreUtil'
import type { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
import './testUtils.scss'
import { PermissionsContext, PermissionsContextProps } from 'framework/rbac/PermissionsContext'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseGetMockDataWithMutateAndRefetch<T> extends UseGetMockData<T> {
  mutate: () => Record<string, unknown>
  refetch: () => Record<string, unknown>
  cancel?: () => Record<string, unknown>
}

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): HTMLElement | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): HTMLElement | null => document.querySelector('.bp3-popover-content')

export interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, unknown>
  defaultAppStoreValues?: Partial<AppStoreContextProps>
  defaultLicenseStoreValues?: Partial<LicenseStoreContextProps>
  defaultPermissionValues?: Partial<PermissionsContextProps>
  defaultFeaturesValues?: Partial<FeaturesContextProps>
  projects?: Project[]
  enableBrowserView?: boolean
  stringsData?: Record<string, string>
  getString?(key: string): string
}

export const prependAccountPath = (path: string): string => withAccountId(() => path)(accountPathProps)

export const CurrentLocation = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export interface BrowserViewProps {
  enable?: boolean
  children: React.ReactNode
}

export function BrowserView(props: BrowserViewProps): React.ReactElement {
  const { enable, children } = props
  const location = useLocation()
  const history = useHistory()

  if (!enable) {
    return <>{children}</>
  }

  function handlePathChange(e: React.ChangeEvent<HTMLInputElement>): void {
    history.replace(e.currentTarget.value)
  }

  const search = location.search ? `?${location.search.replace(/^\?/, '')}` : ''

  return (
    <div className="browser">
      <div className="browser-header">
        <input className="browser-path" value={location.pathname + search} onChange={handlePathChange} />
      </div>
      <div className="browser-content">{children}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  enableMapSet()
  const {
    path = '/',
    pathParams = {},
    defaultAppStoreValues,
    queryParams = {},
    defaultLicenseStoreValues,
    defaultPermissionValues,
    defaultFeaturesValues,
    stringsData = {},
    getString = (key: string) => key
  } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const history = React.useMemo(() => createMemoryHistory({ initialEntries: [routePath] }), [])

  const defaultReturn = {
    enabled: true
  }

  /** TODO: Try fixing this later. This is causing some tests to fail */
  // React.useEffect(() => {
  //   history.replace(compile(path)(pathParams) + qs.stringify(queryParams, { addQueryPrefix: true }))
  //   // eslint-disable-next-line react-hooks/exhaustive-deps
  // }, [path, pathParams, queryParams])

  return (
    <StringsContext.Provider value={{ data: stringsData as any, getString }}>
      <AppStoreContext.Provider
        value={{
          featureFlags: {
            FEATURE_ENFORCEMENT_ENABLED: true
          },
          updateAppStore: () => void 0,
          currentUserInfo: {},
          ...defaultAppStoreValues
        }}
      >
        <LicenseStoreContext.Provider
          value={{
            versionMap: {},
            licenseInformation: {},
            CI_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            FF_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CCM_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CD_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            updateLicenseStore: () => void 0,
            ...defaultLicenseStoreValues
          }}
        >
          <PermissionsContext.Provider
            value={{
              permissions: new Map<string, boolean>(),
              requestPermission: () => void 0,
              checkPermission: () => true,
              cancelRequest: () => void 0,
              ...defaultPermissionValues
            }}
          >
            <FeaturesContext.Provider
              value={{
                features: new Map<FeatureIdentifier, FeatureDetail>(),
                featureMap: new Map<FeatureIdentifier, FeatureMetaData>(),
                getEdition: () => {
                  return undefined
                },
                requestFeatures: () => void 0,
                requestLimitFeature: () => void 0,
                checkFeature: () => {
                  return defaultReturn
                },
                checkLimitFeature: () => {
                  return defaultReturn
                },
                getRestrictionType: () => {
                  return undefined
                },
                ...defaultFeaturesValues
              }}
            >
              <Router history={history}>
                <ModalProvider>
                  <RestfulProvider base="/">
                    <BrowserView enable={props.enableBrowserView}>
                      <Switch>
                        <Route exact path={path}>
                          {props.children}
                        </Route>
                        <Route>
                          <CurrentLocation />
                        </Route>
                      </Switch>
                    </BrowserView>
                  </RestfulProvider>
                </ModalProvider>
              </Router>
            </FeaturesContext.Provider>
          </PermissionsContext.Provider>
        </LicenseStoreContext.Provider>
      </AppStoreContext.Provider>
    </StringsContext.Provider>
  )
}

export const queryByNameAttribute = (name: string, container: HTMLElement): HTMLElement | null =>
  queryByAttribute('name', container, name)
