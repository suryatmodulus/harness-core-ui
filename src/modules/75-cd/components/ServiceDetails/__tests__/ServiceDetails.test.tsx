/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import ServiceDetails from '@cd/components/ServiceDetails/ServiceDetails'
import * as cdngServices from 'services/cd-ng'

jest.mock('highcharts-react-official', () => () => <></>)
jest.mock('@common/utils/YamlUtils', () => ({}))

jest.spyOn(cdngServices, 'useGetActiveServiceInstanceSummary').mockImplementation(() => {
  return { loading: false, error: false, data: [], refetch: jest.fn() } as any
})

jest.spyOn(cdngServices, 'useGetEnvBuildInstanceCount').mockImplementation(() => {
  return {
    mutate: () => Promise.resolve({ loading: true, data: [] })
  } as any
})

jest.spyOn(cdngServices, 'useGetServiceDeploymentsInfo').mockImplementation(() => {
  return { loading: false, error: false, data: [], refetch: jest.fn() } as any
})

jest.spyOn(cdngServices, 'useGetInstanceGrowthTrend').mockImplementation(() => {
  return { loading: false, error: false, data: [], refetch: jest.fn() } as any
})

jest.spyOn(cdngServices, 'useGetInstanceCountHistory').mockImplementation(() => {
  return { loading: false, error: false, data: [], refetch: jest.fn() } as any
})

jest.spyOn(cdngServices, 'useGetDeploymentsByServiceId').mockImplementation(() => {
  return { loading: false, error: false, data: [], refetch: jest.fn() } as any
})

jest.spyOn(cdngServices, 'useGetServiceHeaderInfo').mockImplementation(() => {
  return { loading: false, error: false, data: [], refetch: jest.fn() } as any
})

describe('ServiceDetails', () => {
  test('should render ServiceDetails', () => {
    const { container } = render(
      <TestWrapper
        path="account/:accountId/cd/orgs/:orgIdentifier/projects/:projectIdentifier/services"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <ServiceDetails />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
