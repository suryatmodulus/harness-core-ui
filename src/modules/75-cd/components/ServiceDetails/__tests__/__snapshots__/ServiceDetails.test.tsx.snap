// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceDetails should render ServiceDetails 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--large header StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="PageHeader--breadcrumbsDiv"
        >
          <div
            class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/dummy/settings"
              >
                <span
                  class="bp3-icon bp3-icon-cog StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-primary7"
                  icon="cog"
                >
                  <svg
                    data-icon="cog"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      cog
                    </desc>
                    <path
                      d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  common.accountSettings
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/dummy/undefined/orgs/dummy/projects/dummy/services"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  services
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main tabsContainer StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge"
    >
      <div
        class="bp3-tabs Tabs--main Tabs--horizontal"
      >
        <div
          class="bp3-tab-list"
          role="tablist"
        >
          <div
            class="bp3-tab-indicator-wrapper"
            style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
          >
            <div
              class="bp3-tab-indicator"
            />
          </div>
          <div
            aria-controls="bp3-tab-panel_serviceDetailsTab_summaryTab"
            aria-disabled="false"
            aria-expanded="true"
            aria-selected="true"
            class="bp3-tab"
            data-tab-id="summaryTab"
            id="bp3-tab-title_serviceDetailsTab_summaryTab"
            role="tab"
            tabindex="0"
          >
            summary
          </div>
          <div
            aria-controls="bp3-tab-panel_serviceDetailsTab_refrencedByTab"
            aria-disabled="false"
            aria-expanded="false"
            aria-selected="false"
            class="bp3-tab"
            data-tab-id="refrencedByTab"
            id="bp3-tab-title_serviceDetailsTab_refrencedByTab"
            role="tab"
            tabindex="0"
          >
            refrencedBy
          </div>
        </div>
        <div
          aria-hidden="false"
          aria-labelledby="bp3-tab-title_serviceDetailsTab_summaryTab"
          class="bp3-tab-panel"
          id="bp3-tab-panel_serviceDetailsTab_summaryTab"
          role="tabpanel"
        >
          <div
            class="StyledProps--font StyledProps--main PageBody--pageBody"
            data-name="page-body"
          >
            <div
              class="StyledProps--font StyledProps--main timeRangeContainer StyledProps--flex"
            >
              <span
                class="bp3-popover-wrapper timeRangeSelector nonMinimalStyles"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main timeRangeButton StyledProps--font StyledProps--font-size-xsmall Button--with-current-color Button--without-shadow Button--withRightIcon"
                    style="height: 20px;"
                    type="button"
                  >
                    <span
                      class="bp3-button-text"
                    >
                      common.duration.month
                    </span>
                    <span
                      class="bp3-icon bp3-icon-caret-down StyledProps--main"
                      icon="caret-down"
                    >
                      <svg
                        data-icon="caret-down"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <desc>
                          caret-down
                        </desc>
                        <path
                          d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
            <div
              class="Layout--horizontal StyledProps--main StyledProps--margin-top-large StyledProps--margin-bottom-large"
            >
              <div
                class="Layout--vertical StyledProps--main StyledProps--margin-right-xlarge"
              >
                <div
                  class="Layout--horizontal StyledProps--main StyledProps--margin-bottom-medium"
                >
                  <div
                    class="bp3-card bp3-elevation-0 Card--card activeServiceInstances"
                  >
                    <div
                      class="Layout--vertical StyledProps--main"
                      style="height: 100%;"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey600 StyledProps--margin-bottom-medium"
                      >
                        cd.serviceDashboard.activeServiceInstancesLabel
                      </p>
                      <div
                        class="Layout--horizontal StyledProps--main activeServiceInstancesHeader StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-space-between StyledProps--padding-bottom-small"
                      >
                        <div
                          class="Layout--horizontal StyledProps--main"
                        >
                          <p
                            class="StyledProps--font StyledProps--main instanceCount StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey800 StyledProps--margin-right-large"
                          >
                            0
                          </p>
                          <div
                            class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-flex-start"
                          >
                            <div
                              class="tickerContainer"
                              data-test="ticker"
                            >
                              <div
                                class="ticker"
                              >
                                <div
                                  class="iconContainer"
                                >
                                  <span
                                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-red500"
                                    data-icon="main-caret-up"
                                  >
                                    <svg
                                      height="6"
                                      viewBox="0 0 14 9"
                                      width="6"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M13.7 7.2L7.9.4C7.7.2 7.4 0 7 0s-.7.2-.9.4L.3 7.2c-.2.2-.3.4-.3.7C0 8.5.5 9 1.2 9h11.7c.6 0 1.1-.5 1.1-1.1 0-.3-.1-.5-.3-.7z"
                                        fill="#4d5969"
                                      />
                                    </svg>
                                  </span>
                                </div>
                                <div
                                  class="tickerValue"
                                  data-test="tickerValue"
                                >
                                  <p
                                    class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--color StyledProps--color-red500"
                                  >
                                    0%
                                  </p>
                                </div>
                              </div>
                            </div>
                            <p
                              class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
                            >
                              cd.serviceDashboard.in
                            </p>
                          </div>
                        </div>
                        <div
                          class="Layout--horizontal StyledProps--main"
                        />
                      </div>
                      <div
                        class="Layout--vertical StyledProps--main activeServiceInstancesEmpty StyledProps--flex StyledProps--flex-align-center-center"
                        data-test="ActiveServiceInstancesEmpty"
                        style="height: 360px;"
                      >
                        <div
                          class="StyledProps--font StyledProps--main StyledProps--margin-bottom-medium"
                        >
                          <img
                            height="50"
                            src="test-file-stub"
                            style="align-self: center;"
                            width="50"
                          />
                        </div>
                        <p
                          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                        >
                          cd.serviceDashboard.noActiveServiceInstances
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="bp3-card bp3-elevation-0 Card--card instanceCountHistory"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600 StyledProps--margin-bottom-small"
                  >
                    cd.serviceDashboard.instanceCountHistory
                  </p>
                  <div
                    class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
                    style="height: 100%;"
                  >
                    <div
                      class="StyledProps--font StyledProps--main StyledProps--margin-bottom-medium"
                    >
                      <img
                        height="50"
                        src="test-file-stub"
                        style="align-self: center;"
                        width="50"
                      />
                    </div>
                    <p
                      class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                    >
                      cd.serviceDashboard.noServiceInstances
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="Layout--vertical StyledProps--main fullWidth"
              >
                <div
                  class="bp3-card bp3-elevation-0 Card--card card"
                >
                  <div
                    class="bp3-card bp3-elevation-0 Card--card card"
                  >
                    <div
                      class="Layout--vertical StyledProps--main"
                      style="height: 100%;"
                    >
                      <div
                        class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
                        data-test="deploymentsWidgetEmpty"
                        style="height: 100%;"
                      >
                        <img
                          height="100"
                          src="test-file-stub"
                          style="align-self: center;"
                          width="150"
                        />
                        <p
                          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--margin-top-medium"
                        >
                          cd.serviceDashboard.noDeployments
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="StyledProps--font StyledProps--main StyledProps--padding-top-medium"
                  style="height: 100%;"
                >
                  <div
                    class="Layout--vertical StyledProps--main"
                    style="height: 100%;"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main StyledProps--padding-top-medium StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-space-between"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey600"
                      >
                        cd.serviceDashboard.totalPipelines: 0
                      </p>
                      <div
                        class="bp3-input-group ui-search-box ExpandingSearchInput--main  ExpandingSearchInput--flip  ExpandingSearchInput--light"
                        data-name=""
                        style="max-width: 200px;"
                      >
                        <span
                          class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
                          data-icon="thinner-search"
                        >
                          <svg
                            fill="none"
                            height="14"
                            viewBox="0 0 16 16"
                            width="14"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                            />
                          </svg>
                        </span>
                        <input
                          class="bp3-input  ExpandingSearchInput--filter"
                          dir="auto"
                          placeholder="search"
                          style="padding-right: 0px;"
                          type="search"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="StyledProps--font StyledProps--main executionCardContainer"
                    >
                      <div
                        class="bp3-card bp3-elevation-0 Card--card pipelineExecutionsEmptyStateContainer"
                      >
                        <div
                          class="StyledProps--font StyledProps--main NoDataCard--noDataCard dataCard StyledProps--flex StyledProps--flex-align-center-center"
                        >
                          <div
                            class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
                            style="align-items: center; margin-top: -48px; width: 470px;"
                          >
                            <img
                              class="pipelineExecutionsEmptyStateImage"
                              src="test-file-stub"
                            />
                            <p
                              class="StyledProps--font StyledProps--main NoDataCard--message StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
                            >
                              cd.serviceDashboard.noPipelines
                            </p>
                            <div
                              class="NoDataCard--buttonContainer"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
